docs/DEV-GUIDE.md

```diff
@@ -0,0 +1,121 @@
+# Xcube Developer Guide
+
+Version 0.1, draft
+
+*IMPORTANT NOTE: Any changes to this doc must be reviewed by dev-team through pull requests.* 
+
+## Index
+
+TODO
+
+## Adding new functionality
+
+Checklist:
```

Comment:
"Checklist" sounds as if all these steps need to be taken in the specified order. I don't think this is entirely the case here.

Reply:
Yes! Please help!